e2e/tests-dfx/install.bash

```diff
@@ -81,3 +81,19 @@ teardown() {
     dfx canister create --all
     assert_command_fail dfx canister install --all --wasm "${archive:?}/wallet/0.10.0/wallet.wasm"
 }
+
+@test "install runs post-install tasks" {
+    dfx_start
+    # shellcheck disable=SC2094
+    cat <<<"$(jq '.canisters.e2e_project."post-install"="sh -c \"echo hello\""' dfx.json)" >dfx.json
+
+    assert_command dfx canister create --all
+    assert_command dfx build
+
+    assert_command dfx canister install --all
```

Comment:
Can we have e2e tests demonstrating expected environment variables are provided for dependencies in the following cases?
- `dfx install --all`
- `dfx install <specific canister>`
- `dfx deploy` (all)
- `dfx deploy <specific canister>`